src/Dialect/Mlir/DialectBuilder.cpp

```diff
@@ -1481,13 +1505,74 @@ Value VectorBuilder::mergeLow(Value lhs, Value rhs, int64_t step) const {
   return shuffle(lhs, rhs, mask);
 }
 
+Value VectorBuilder::reduction(
+    VectorBuilder::CombiningKind kind, Value value) const {
+  Type type = value.getType();
+  switch (kind) {
+  case CombiningKind::ADD: {
+    return b().create<vector::ReductionOp>(
+        loc(), vector::CombiningKind::ADD, value);
+  }
+  case CombiningKind::MUL: {
+    return b().create<vector::ReductionOp>(
+        loc(), vector::CombiningKind::MUL, value);
+  }
+  case CombiningKind::MAX: {
+    if (MathBuilder::isUnsignedIntegerWithVector(type))
+      return b().create<vector::ReductionOp>(
+          loc(), vector::CombiningKind::MAXUI, value);
+    if (MathBuilder::isIntegerWithVector(type))
+      return b().create<vector::ReductionOp>(
+          loc(), vector::CombiningKind::MAXSI, value);
```

Comment:
So if the input type is `tensor<10xi64>` we will use signed integers? Could you check with a simple example of reducemax to see if it really works or not? Just wanted to confirm.

Reply:
added an int test, verified with the new `CheckONNXModel.py -m reducemeanint.mlir -t "-O3 -march=arm64"    ` command... really useful